### ISSUE TITLE
Yolo loss always zero when using batch size 1

### ISSUE THREAD

**Issue by DavideACucci** [OlafenwaMoses/ImageAI#509]
I have the following when using batch size = 1 (loss = 0.0000e+00 for all batches and for all epochs)

```
1/1224 [..............................] - ETA: 11:37:36 - loss: 0.0000e+00 - yolo_layer_1_loss: 0.0000e+00 - yolo_layer_2_loss: 0.0000e+00 - yolo_layer_3_loss: 0.0000e+00
2/1224 [..............................] - ETA: 5:58:09 - loss: 0.0000e+00 - yolo_layer_1_loss: 0.0000e+00 - yolo_layer_2_loss: 0.0000e+00 - yolo_layer_3_loss: 0.0000e+00 
3/1224 [..............................] - ETA: 4:04:44 - loss: 0.0000e+00 - yolo_layer_1_loss: 0.0000e+00 - yolo_layer_2_loss: 0.0000e+00 - yolo_layer_3_loss: 0.0000e+00
4/1224 [..............................] - ETA: 3:06:50 - loss: 0.0000e+00 - yolo_layer_1_loss: 0.0000e+00 - yolo_layer_2_loss: 0.0000e+00 - yolo_layer_3_loss: 0.0000e+00
```

The problem disappears if I use batch size = 2. Is this some kind of error computing the loss?

I would like to use gradient accumulation techniques, e.g. [this one](https://github.com/run-ai/runai/tree/master/runai/ga) to cope with limited GPU memory.

**Comment by AbimbolaOO** [comment]
Hello @DavideACucci,

Firstly, imageAI doesn't support gradient accumulation techniques. 
secondly, concerning the batch size of 1 you are expected to always use batch sizes that of powers of 2 i.e 2, 4, 8, 16, . . . also using you are introducing very high variance into your train data by using a batch size of 1 and you might need to use other kinds of optimizers that are not currently available on imageAI for things to work.

**Comment by DavideACucci** [comment]
Dear @AbimbolaOO,

I know that imageAI does not support gradient accumulation techniques, I wanted to create such extension to be able to train on GPUs with limited memory. Batch size 1 and gradient accumulation should do as far as I know, provided that the loss for the batch with size 1 is properly returned.

Am I wrong?
